### PR TITLE
Run slider animation without using timer

### PR DESCRIPTION
**Pull request by Czaki**
# References and relevant issues

Hope, this will fix test crash in #7146

# Description

This PR simplifies code for play axis slider. The motivation is to reduce the fragility of test s and remove the reason of failing test in #7146

### CODE REVIEW

File: napari/_qt/widgets/_tests/test_qt_play.py [hunk 1]
```diff
@@ -113,7 +114,7 @@ def ref_view(make_napari_viewer):
     viewer = make_napari_viewer()
 
     np.random.seed(0)
-    data = np.random.random((10, 10, 15))
+    data = np.random.random((2, 10, 10, 15))
```
Comment (jni) [inline napari/_qt/widgets/_tests/test_qt_play.py:117]: Why this change @Czaki?

File: napari/_qt/widgets/_tests/test_qt_play.py [hunk 2]
```diff
@@ -155,3 +156,31 @@ def increment(e):
         view.dims.play(2, 20)
     view.dims.dims.events.current_step.disconnect(increment)
     assert not view.dims.is_playing
+
+
+def test_change_play_axis(ref_view, qtbot):
+    """Make sure changing the play axis stops the current animation"""
```
Comment (jni) [inline napari/_qt/widgets/_tests/test_qt_play.py]: Not that we necessarily want to solve the following in this PR, and I'm glad this solves [this crash](https://github.com/napari/napari/pull/6300#issuecomment-1757696072) (and I think it means we can put a bugfix label on this PR!), but I think we would want to be able to do multiple axes simultaneously in the future. So either we remove/change this test or we can add a note in the docstring, like:

```suggestion
def test_change_play_axis(ref_view, qtbot):
    """Make sure changing the play axis stops the current animation.
    
    Prior to https://github.com/napari/napari/pull/7158, starting a new play
    animation crashed Python. In the future, we may want to allow
    multiple simultaneous play axes [1]_, so this test should be removed
    when we do that.
    
    ..[1] https://github.com/napari/napari/pull/6300#issuecomment-1757696072
    """
```

File: napari/_qt/widgets/qt_dims_slider.py [hunk 3]
```diff
@@ -585,43 +533,54 @@ def _handle_stop(self):
...
-
-        slider.fps_changed.connect(self.set_fps)
-        slider.mode_changed.connect(self.set_loop_mode)
-        slider.range_changed.connect(self.set_frame_range)
-        self.set_fps(self.slider.fps)
-        self.set_frame_range(slider.frame_range)
+        self.slider = None
+        # self.dims = slider.dims
+        # self.axis = slider.axis
+        # self.loop_mode = slider.loop_mode
```
Comment (jni) [inline napari/_qt/widgets/qt_dims_slider.py]: ?

File: napari/_qt/widgets/qt_dims_slider.py [hunk 4]
```diff
@@ -642,18 +600,20 @@ def work(self):
                 self.frame_requested.emit(self.axis, self.min_point)
             elif self.step < 0 and self.current <= self.min_point + 1:
                 self.frame_requested.emit(self.axis, self.max_point)
-            self.timer.start()
+            # self.timer.start()
         else:
             # immediately advance one frame
             self.advance()
-        self.started.emit()
+        # next_time = time.time() + steep
```
Comment (jni) [inline napari/_qt/widgets/qt_dims_slider.py]:
```suggestion
        # next_time = time.time() + step
```

But maybe remove the comment?

**Review by jni — APPROVED** [review]
Very nice, I think this is a great improvement @Czaki. We'll need to do some work to update #6300 but this mechanism is much cleaner so I hope that it's easy.

I had a few minor comments and suggestions but I will approve optimistically as this is already a significant improvement.

File: napari/_qt/widgets/_tests/test_qt_play.py (hunk 1 above)
Comment (Czaki) [reply napari/_qt/widgets/_tests/test_qt_play.py:117]: To have two visible sliders

File: napari/_qt/widgets/qt_dims_slider.py (hunk 4 above)
Comment (Czaki) [reply napari/_qt/widgets/qt_dims_slider.py]: removed

File: napari/_qt/widgets/qt_dims_slider.py (hunk 3 above)
Comment (Czaki) [reply napari/_qt/widgets/qt_dims_slider.py]: removed

File: napari/_qt/widgets/_tests/test_qt_play.py (hunk 2 above)
Comment (Czaki) [reply napari/_qt/widgets/_tests/test_qt_play.py]: There was no crash. There was a lack of code coverage

Comment (jni) [reply napari/_qt/widgets/_tests/test_qt_play.py]: @Czaki see the linked comment — if you click play on second axis after starting first axis, you get a crash. I just tried it now.

```
WARNING: QThread: Destroyed while thread is still running
16:50:50 : WARNING : MainThread : QThread: Destroyed while thread is still running
```

Comment (jni) [reply napari/_qt/widgets/_tests/test_qt_play.py]: I just confirmed that with this PR, the first play axis stops playing. This is much preferable to the previous behaviour. 😅

Comment (Czaki) [reply napari/_qt/widgets/_tests/test_qt_play.py]: > @Czaki see the linked comment — if you click play on second axis after starting first axis, you get a crash. I just tried it now.

On my machine it creates warning but napari was still running. And the second click starts the animation again. 
I was my intentional change to require only one click.

Comment (jni) [reply napari/_qt/widgets/_tests/test_qt_play.py]: Ok I have made the message clearer that it only caused crashes in some situations.

```suggestion
def test_change_play_axis(ref_view, qtbot):
    """Make sure changing the play axis stops the current animation.
    
    Prior to https://github.com/napari/napari/pull/7158, starting a new play
    animation resulted in QThread warnings and could crash Python in
    some environments. In the future, we may want to allow multiple
    multiple simultaneous play axes [1]_, so this test should be changed
    or removed when we do that.
    
    ..[1] https://github.com/napari/napari/pull/6300#issuecomment-1757696072
    """
```
